apps/site/assets/ts/stop/components/DepartureTimes.tsx

```diff
@@ -292,18 +297,27 @@ const DepartureTimes = ({
   );
 
   const schedules = schedulesByHeadsign(schedulesForDirection);
-
+  let departures: DepartureInfo[] = [];
+  let id: number = 0;
   return (
     <>
       {Object.entries(schedules).map(([headsign, schs]) => {
-        const formattedTimes = toDisplayTime(
-          schs,
-          predictionsByHeadsign[headsign]
-            ? predictionsByHeadsign[headsign]
-            : [],
-          overrideDate
+        const preds = predictionsByHeadsign[headsign]
+          ? predictionsByHeadsign[headsign]
+          : [];
+        const formattedTimes = toDisplayTime(schs, preds, overrideDate);
+        departures = mergeIntoDepartureInfo(schs, preds);
+        id += 1;
+        return (
+          <div
+            key={`${headsign}-${id}`}
+            onClick={() => onClick(route, directionId, departures)}
```

Comment:
suggestion: It looks like the value for route here would be the same for different headsigns, but I think we would want to pass an identifier for the different headsigns so that clicking each of the rows under the 39 would result in show the departures only associated with that headsign.
![image](https://github.com/mbta/dotcom/assets/31781298/7ce4ad56-5c4a-46d8-a0e6-3620013542cd)

Each headsign corresponds to a [route pattern](https://api-v3.mbta.com/docs/swagger/index.html#/RoutePattern/ApiWeb_RoutePatternController_show). 

I would recommend passing the route_pattern_id instead of the route id here. Or addressing in a separate PR if need be!

Reply:
Based on discussion in slack, I think it is OK for now not to pass route_pattern_id. This can be revisited if needed as part of future implementation work on the DepartureList + realtime map.